### PR TITLE
update python-gevent to latest

### PR DESCRIPTION
**Pull request by tobiasb-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Update `python-gevent` to latest.

###### Change Log  

- Update python-gevent to latest
- Remove now-spurious patches

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Builds locally
- Package Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=458921&view=results

### CODE REVIEW

File: SPECS/python-gevent/python-gevent.spec [hunk 1]
```diff
@@ -64,6 +59,8 @@ nosetests
 %{python3_sitelib}/*
 
 %changelog
+* Mon Nov 27 2023 Tobias Brick <tobiasb@microsoft.com> - 23.9.1-1
+
```
Comment (christopherco) [inline SPECS/python-gevent/python-gevent.spec:63]: Please add 1 line in the changelog entry describing this is an update to 23.9.1

File: SPECS/python-gevent/python-gevent.signatures.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "gevent-1.3.6.tar.gz": "7b413c391e8ad6607b7f7540d698a94349abd64e4935184c595f7cdcc69904c6"
+    "gevent-23.9.1.tar.gz": "6d38bb2a80d2cb08becdf52ebb007770107adbf2f2a66bc8eaa750501b0ca006"
```
Comment (christopherco) [inline SPECS/python-gevent/python-gevent.signatures.json:3]:
```suggestion
  "gevent-23.9.1.tar.gz": "6d38bb2a80d2cb08becdf52ebb007770107adbf2f2a66bc8eaa750501b0ca006"
```
nit

**Review by christopherco — COMMENTED** [review]
Changes themselves LGTM. Just a couple minor comments to address

File: SPECS/python-gevent/python-gevent.spec (hunk 1 above)
Comment (tobiasb-ms) [reply SPECS/python-gevent/python-gevent.spec:63]: I really thought I had. Weird; thanks.

**Review by christopherco — APPROVED** [review]

**Pull request merged**